flink-core/src/main/java/org/apache/flink/types/Row.java

```diff
@@ -66,10 +66,11 @@ public int getArity() {
 	 * Gets the field at the specified position.
 	 * @param pos The position of the field, 0-based.
 	 * @return The field at the specified position.
-	 * @throws IndexOutOfBoundsException Thrown, if the position is negative, or equal to, or larger than the number of fields.
+	 * Return null if the position is equal to, or larger than the number of fields.
+	 * @throws IndexOutOfBoundsException Thrown, if the position is negative.
 	 */
 	public Object getField(int pos) {
-		return fields[pos];
+		return pos >= fields.length ? null : fields[pos];
```

Comment:
Hi @fhueske . Thank you for your comments. The overhead of `if`s are additionally caused here. It may not be an efficient way. IMHO, The design of `TableFunction` support arbitrary types of output Row. However, too many kinds of `TableFunction`s may also confuse users. I will  discover whether there is a concise and efficient way.